editor.planx.uk/src/@planx/components/MapAndLabel/Public/CopyFeature.tsx

```diff
@@ -0,0 +1,59 @@
+import Box from "@mui/material/Box";
+import MenuItem from "@mui/material/MenuItem";
+import { Feature } from "geojson";
+import React from "react";
+import SelectInput from "ui/editor/SelectInput";
+import InputLabel from "ui/public/InputLabel";
+
+import { useMapAndLabelContext } from "./Context";
+
+interface Props {
+  destinationIndex: number;
+  features: Feature[];
+}
+
+export const CopyFeature: React.FC<Props> = ({
+  destinationIndex,
+  features,
+}) => {
+  const { schema, copyFeature } = useMapAndLabelContext();
+
+  // Only show component if there are multiple features
+  if (features.length < 2) return null;
+
+  // We can only copy from other features
+  const sourceFeatures = features.filter(
+    (_, sourceIndex) => sourceIndex !== destinationIndex,
+  );
+
+  return (
+    <Box>
+      <InputLabel label="Copy from" id={`select-${destinationIndex}`}>
+        <SelectInput
+          bordered
+          required
+          title={"Copy from"}
+          labelId={`select-label-${destinationIndex}`}
+          value={""}
+          onChange={(e) => {
+            const label = e.target.value as string;
+            // Convert text label to zero-indexed integer
+            const sourceIndex = parseInt(label, 10) - 1;
+            copyFeature(sourceIndex, destinationIndex);
+          }}
```

Comment:
This feels a little more complex than it needs to, but feels like the most readable version of what I'm aiming for.

If we use indices, it falls apart as we're "skipping" the current feature in the select. The alternatives are either something horrible like `value={ j > i ? j + 1 : i}` in the select to dynamically "skip", or keeping the current item in the select and making it disabled (not great for a11y).